zenodo/modules/github/views/badge.py

```diff
@@ -104,18 +104,23 @@ def doi_badge(doi):
 
     if pid is None:
         return abort(404)
-    return badge(doi)
+    style = request.args.get('style', None)
+
+    return badge(doi, style)
 
 
 @blueprint.route("/<int:user_id>/<path:repository>.png", methods=["GET"])
 @ssl_required
 def index_old(user_id, repository):
     """Legacy support for old badge icons."""
-    return redirect(url_for('.index', user_id=user_id, repository=repository))
+    full_url = url_for('.index', user_id=user_id,
+                       repository=repository, **request.args)
```

Comment:
``request.args`` contains unvalidated data so it should not be passed blindly. Better just extract style, and pass it on.